src/big_widgets/GitQlient.cpp

```diff
@@ -110,67 +102,82 @@ void GitQlient::setRepositories(const QStringList &repositories)
       addRepoTab(repo);
 }
 
-void GitQlient::setArgumentsPostInit(const QStringList &arguments)
+bool GitQlient::setArgumentsPostInit(const QStringList &arguments)
 {
    QLog_Info("UI", QString("External call with the params {%1}").arg(arguments.join(",")));
 
-   const auto repos = parseArguments(arguments);
-
-   setRepositories(repos);
+   QStringList repos;
+   bool ret = parseArguments(arguments, &repos);
+   if (ret) {
+      setRepositories(repos);
+   }
+   return ret;
 }
 
-QStringList GitQlient::parseArguments(const QStringList &arguments)
+bool GitQlient::parseArguments(const QStringList &arguments, QStringList *repos)
```

Comment:
I'm curious about why did you use a pointer here instead of passing by reference.

Reply:
To make it obvious at call site that this is out parameter.